re_data_ui/src/utils/graph.ts

```diff
@@ -24,13 +30,19 @@ const getLayoutElements = (elements: Elements, direction = 'LR'): Elements => {
     } else {
       dagreGraph.setEdge(element.source, element.target);
     }
-  });
+  }
 
   dagre.layout(dagreGraph);
 
-  return elements.map((el) => {
-    const element = el as Node;
+  const newElements = [];
+  const elementsObj: Record<string, number> = {};
+
+  for (let index = 0; index < elements.length; index++) {
+    const element = elements[index] as Node;
+
     if (isNode(element)) {
+      elementsObj[element.data?.id] = index;
```

Comment:
isn't there a possibility of this resolving to
```
elementsObj[undefined] = index;
```

since you're using optional chaining.